pkg/netlinksafe/netlink.go

```diff
@@ -0,0 +1,321 @@
+// Package netlinksafe wraps vishvandanda/netlink functions that may return EINTR.
```

Comment:
if you forking the code it is better to indicate the source , open source licenses and copyright  should be respected

License is apache so it is ok, but I noticed they have copyright https://github.com/moby/moby/blob/6de8ba3bc52377ddb06d80c76b0c8d302cb81261/LICENSE#L179

// Copyright Docker, Inc.
// Copied from https://github.com/moby/moby/blob/edaa0eb56d7e749df81bd3dcc2945b81e911b52e/internal/nlwrap/nlwrap_linux.go

Different projects follow different rules, but this seems to be correct to the best of my knowledge

Reply:
Right, a copyright attribution is the right thing to do here. Something like

```
// SPDX-License-Identifier: Apache-2.0
// Copyright Docker, Inc.
// Copied from https://github.com/moby/moby/blob/edaa0eb56d7e749df81bd3dcc2945b81e911b52e/internal/nlwrap/nlwrap_linux.go
```

Given that the upstream files do not have a copyright header, we're on our own to Do The Right Thing.